data-runtime/src/main/java/io/micronaut/data/runtime/intercept/AbstractQueryInterceptor.java

```diff
@@ -108,6 +109,47 @@ protected AbstractQueryInterceptor(@NonNull RepositoryOperations operations) {
         this.operations = operations;
     }
 
+    /**
+     * Returns the return type.
+     *
+     * @param context The context
+     * @return the return type
+     */
+    protected Argument<?> getReturnType(MethodInvocationContext<?, ?> context) {
+        return context.getReturnType().asArgument();
+    }
+
+    @Nullable
+    protected final Object convertOne(MethodInvocationContext<?, ?> context, @Nullable Object o) {
+        Argument<?> argumentType = getReturnType(context);
+        Class<?> type = argumentType.getType();
+        if (o == null) {
+            if (type == Optional.class) {
+                return Optional.empty();
+            }
+            if (argumentType.isDeclaredNonNull() || !argumentType.isNullable()
+                    && !context.getReturnType().asArgument().isNullable()) {
```

Comment:
this can be replaced with `argumentType`

Reply:
I'm trying to check if something can be null and I have cases like `Future<Type>` where `argumentType` is `Type` and the last check is for the future.